### ISSUE TITLE
VRS add concept of variant rareness

### ISSUE THREAD

**Issue by torbjorgen** [Clinical-Genomics/rdds#156]
# Rare Variant Model Tuning

An expectation for novel variants is that
theyre less annotated than previously known variants.

See [exploration notebook notes](https://github.com/Clinical-Genomics/rdds/blob/622fb6e0ef34720dcf4d4b92c5fbc49b8d2916ec/src/rdds/variant_rank_score/inference_exploration/explore_patient_cases.ipynb):

> 
    Variants that are less annotated are more likely to have an (incorrect) benign prediction
    Variants that are not annotated with CLINVAR_CLNSIG are more likely to be incorrectly classified as benign
    It's mainly the text features that's driving the sparsity issue, judging from the network plot for textual features.
    Furthermore, missing spliceAI annotations seems to be correlated with low pathogenicity score
    It might be worth syntesizing variants that are sparser in the following annotations, during training:
        CLINVAR_CLNSIG|CLNREVSTAT
        GERP++_NR|_RS
        SpliceAI_*
        PhastCons100way_vertebrate
        (PhyloP100way_vertebrate)
        REVEL_score
        SIFT
        SPIDEX

## NaN frequencies in Patient Case Data (n=30)

![Image](https://github.com/user-attachments/assets/4ba28060-be03-4328-aef3-f3694210ac5e)

## Actions
Add this information to the training in the loss function?

Add frq into the loss function, to score rare variants higher.

## Postprocessing Changes
Select only high-scored and rare variants as output from the model.

## Model Changes
Look into TF Lattice and add constraint for
observed frequencies as a indicator for rareness.

https://www.tensorflow.org/lattice/overview#interpretability

Plot model score vs annotation amount to see if they're dependant.
